src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -48,208 +49,168 @@
-    private final Set<BlockPos>                   freePositions     = new HashSet<>();
+    private boolean manualHiring = false;
     /**
-     * The Blocks which players can freely interact with.
+     * The housing mode in the colony.
+     */
+    private boolean manualHousing = false;
+    /**
+     * The name of the colony.
+     */
+    private String name = "ERROR(Wasn't placed by player)";
+    /**
+     * The center of the colony.
+     */
+    private BlockPos center;
+    /**
+     * The colony permission object.
      */
-    private final Set<Block>                      freeBlocks        = new HashSet<>();
-    private       int                             minedOres         = 0;
-    private       int                             minedDiamonds     = 0;
-    private       int                             harvestedWheat    = 0;
-    private       int                             harvestedPotatoes = 0;
-    private       int                             harvestedCarrots  = 0;
-    private       int                             killedMobs        = 0;
-    private       int                             builtHuts         = 0;
-    private       int                             caughtFish        = 0;
-    private       int                             felledTrees       = 0;
-    private       int                             plantedSaplings   = 0;
-    //  Runtime Data
-    @Nullable
-    private       World                           world             = null;
-    //  Updates and Subscriptions
-    @NotNull
-    private       Set<EntityPlayerMP>             subscribers      = new HashSet<>();
-    private       boolean                         isDirty          = false;
-    private       boolean                         isCitizensDirty  = false;
-    private       boolean                         isBuildingsDirty = false;
-    private       boolean                         manualHiring     = false;
-    private       boolean                         manualHousing     = false;
-
-    private       boolean                         isFieldsDirty    = false;
-    private       String                          name             = "ERROR(Wasn't placed by player)";
-    private BlockPos         center;
-    //  Administration/permissions
     @NotNull
-    private Permissions      permissions;
+    private Permissions permissions;
+    /**
+     * The townhall of the colony.
+     */
     @Nullable
     private BuildingTownHall townHall;
+    /**
+     * The highest citizen id.
+     */
     private int topCitizenId = 0;
-    private int maxCitizens  = Configurations.gameplay.maxCitizens;
-
-    private double overallHappiness = 5;
-
     /**
-     * Colony permission event handler.
+     * Max citizens without housing.
      */
-    private final ColonyPermissionEventHandler eventHandler;
-
+    private int maxCitizens = Configurations.gameplay.maxCitizens;
+    /**
+     * Overall happyness of the colony.
+     */
+    private double overallHappiness = 5;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)